central/networkgraph/flow/datastore/internal/store/postgres/store.go

```diff
@@ -613,6 +665,44 @@ func (s *flowStoreImpl) RemoveOrphanedFlows(ctx context.Context, orphanWindow *t
 	return s.pruneFlows(ctx, pruneStmt, orphanWindow)
 }
 
+func (s *flowStoreImpl) pruneOrphanExternalEntities(ctx context.Context, srcFlows []*storage.NetworkFlow, dstFlows []*storage.NetworkFlow) error {
+	// srcFlows contains flows where src is the deployment,
+	// so prune external flows based on the dst entity
```

Comment:
```suggestion
	// so prune external flows based on the dst entity
	s.mutex.Lock()
	defer s.mutex.Unlock()	
```
You need to make sure that this doesn't interfere with other operation that might be done on the flows and entities tables.

Reply:
There is already a mutex in the call to `RemoveOrphanedFlows`.

Reply:
Oops. Yes, that is correct. However, I think the mutex shouldn't actually be locked the entire time. I think it should probably be locked and unlocked multiple times, so that it doesn't timeout and to give other queries a chance to modify these tables.

Reply:
I have done this in another PR https://github.com/stackrox/stackrox/pull/15008